data/dataset.py

```diff
@@ -112,8 +121,77 @@ def _build_dataset(self) -> tuple[list[dict], list[dict]]:
 
         return processed_records, unprocessed_records
 
-    def _build_vocab(self) -> tuple[list[str], list[str]]:
-        vocab_src, vocab_tgt = ["<s>", "</s>"], ["<s>", "</s>"]
+    def build_vocab(self) -> tuple[list[str], list[str]]:
+        vocab_src, vocab_tgt = ["<s>", "<blank>", "</s>"], ["<s>", "<blank>", "</s>"]
+        iterators = {
+            "pitch": range(21, 109),
+            "duration_bin": range(self.quantizer.n_duration_bins),
+            "dstart_bin": range(self.quantizer.n_dstart_bins),
+            "velocity_bin": range(self.quantizer.n_velocity_bins),
+        }
+
+        src_product = itertools.product(
+            iterators[self.tokenizer_src.keys[0]],
+            iterators[self.tokenizer_src.keys[1]],
+        )
+
+        tgt_product = itertools.product(
+            iterators[self.tokenizer_tgt.keys[0]],
+            iterators[self.tokenizer_tgt.keys[1]],
+        )
+
+        for val1, val2 in src_product:
+            key = f"{val1}-{val2}"
+            vocab_src.append(key)
+        for val1, val2 in tgt_product:
+            key = f"{val1}-{val2}"
+            vocab_tgt.append(key)
+
+        return vocab_src, vocab_tgt
+
+    def __getitem__(self, idx: int):
+        return self.samples[idx]
+
+    def __len__(self):
+        return len(self.samples)
+
+
+class BinsToVelocityDataset(TokenizedMidiDataset):
+    def __init__(
+        self,
+        split: str = "train",
+        n_dstart_bins: int = 3,
+        n_duration_bins: int = 3,
+        n_velocity_bins: int = 3,
+        sequence_len: int = 128,
+        device: str = "cpu",
```

Comment:
```suggestion
```

We should be able to manage devices outside of datasets :)

Reply:
I do not really know which way would be the best - should I implement "to" method which would iterate throug all the samples and put them on GPU?

Reply:
I usually change the device in the training loop:

```python
for batch in loader:
    x = batch["x"].to(device)
```

Reply:
okay thanks! :))